packages/http/src/validator/validators/utils.ts

```diff
@@ -29,9 +29,10 @@ export const validateAgainstSchema = (value: any, schema: JSONSchema, prefix?: s
     const validate = ajv.compile(schema);
     const valid = validate(value);
     if (!valid) {
-      return convertAjvErrors(validate.errors, DiagnosticSeverity.Error).map(error => {
+      const enhacedMessages = betterAjvErrors(schema, value, validate.errors, { format: 'js' });
+      return convertAjvErrors(validate.errors, DiagnosticSeverity.Error).map((error, index) => {
         const path = prefix ? [prefix, ...error.path] : [...error.path];
-        return Object.assign({}, error, { path });
+        return Object.assign({}, error, { path, message: enhacedMessages[index].error });
```

Comment:
```suggestion
        return { ...error, path, message: enhancedMessages[index].error };
```
Note `enhaced` -> `enhanced`